packages/ember-data/lib/system/record-array-manager.js

```diff
@@ -101,6 +107,13 @@ export default Ember.Object.extend({
       filter = get(array, 'filterFunction');
       this.updateRecordArray(array, filter, typeClass, record);
     }, this);
+
+    if (this.liveRecordArrays.has(typeClass)) {
+      var liveRecordArray = this.liveRecordArrays.get(typeClass);
+      if (!liveRecordArray.isDestroyed && !liveRecordArray.isDestroying) {
```

Comment:
it seems like previously it was handled by unregisterFilteredRecordArray, called when in the recordArray's destroy method.
Maybe this could be rename as unregisterRecordArray which could handle both filtered/live record arrays ?